chore: Write read me for the Markdoc config. (#7989)
withastro/astro: packages/integrations/markdoc/README.md

📚 [See the Markdoc documentation](https://markdoc.dev/docs/functions#creating-a-custom-function) for more on using variables or functions in your content.

### Markdoc Language Server

If you are using VS Code, there is a [Markdoc language extension](https://marketplace.visualstudio.com/items?itemName=Stripe.markdoc-language-support). Which includes syntax highlighting and autocomplete for configured tags.
For more information, [see the language server on GitHub](https://github.com/markdoc/language-server.git).

To set up the extension, create a `markdoc.config.json` file into the project root, with following content:

```json
[
  {
    "id": "my-site",
    "path": "src/content",
    "schema": {
      "path": "markdoc.config.mjs",
      "type": "esm",
      "property": "default",
      "watch": true
    },
    "routing": {
      "frontmatter": "route"
    }
  }
]
```

> The `schema` property, is the most important. It contains all information to configure the language server.
> It accepts following properties.
> 
> - `path`: The path to the configuration file.
> - `type`: Kind of module your configuration file uses.
> - `property`: The exported property name that contains the configuration object.
> - `watch`: Tell the server to watch for changes in the configuration.
> 
> The global `path` property, tells the server where the content is located. You can't use a glob pattern. 

Reviewer: ```suggestion
The `schema` property contains all information to configure the language server for Astro content collections.
It accepts following properties:

- `path`: The path to the configuration file.
- `type`: The type of module your configuration file uses (`esm` allows `import` syntax).
- `property`: The exported property name that contains the configuration object.
- `watch`: Tell the server to watch for changes in the configuration.

The top-level `path` property tells the server where content is located. Since Markdoc is specific to content collections, you can use `src/content`.
```

Correcting some improper comma usage and sentence fragments. I also rephrased the "global `path`" section to use "top-level `path`", since I was unsure what global meant at first.

📚 [See the Markdoc documentation](https://markdoc.dev/docs/functions#creating-a-custom-function) for more on using variables or functions in your content.

### Markdoc Language Server

> If you are using VS Code there is a extension called [Markdoc language support](https://marketplace.visualstudio.com/items?itemName=Stripe.markdoc-language-support). It gives you syntax highlighting for markdown and autocomplete for configured tags.
> For more information please go click [here](https://github.com/markdoc/language-server.git).

Reviewer 2: ```suggestion
If you are using VS Code, there is a [Markdoc language extension](https://marketplace.visualstudio.com/items?itemName=Stripe.markdoc-language-support). Which includes syntax highlighting and autocomplete for configured tags.
For more information, [see the language server on GitHub](https://github.com/markdoc/language-server.git).
```

> To configure the language server to work. Create a `markdoc.config.json` file into the project root.
> Then copy the code below and paste it in that file.

Reviewer 2: ```suggestion
To set up the extension, create a `markdoc.config.json` file into the project root, with following content:
```

```json
[
  {
    "id": "my-site",
    "path": "src/content",
    "schema": {
      "path": "markdoc.config.mjs",
      "type": "esm",
      "property": "default",
      "watch": true
    },
    "routing": {
      "frontmatter": "route"
    }
  }
]
```

> If you need to change anything remember that the `schema` property is the one that is important.
> It tells the server information about the config file you are using. 
> It accepts these properties.

Reviewer 2: ```suggestion
The `schema` property, is the most important. It contains all information to configure the language server.
It accepts following properties.
```

> - The path is where the config file is
> - The type is that kind of module the config is using
> - The property is what exported property contains the config object you are using.
> - The watch is to tell the server to watch for changes to the config   

Reviewer 2: ```suggestion
- `path`: The path to the configuration file.
- `type`: Kind of module your configuration file uses.
- `property`: The exported property name that contains the configuration object.
- `watch`: Tell the server to watch for changes in the configuration.
```

> The path that is not in the schema tells the server what folder to watch for.
> **Don't use a glob pattern**. 

Reviewer 2: ```suggestion
The global `path` property, tells the server where the content is located. You can't use a glob pattern. 
```
